ext/session/session.c

```diff
@@ -1686,15 +1700,56 @@ static PHP_FUNCTION(session_set_cookie_params)
 		RETURN_FALSE;
 	}
 
-	convert_to_string_ex(lifetime);
+	if (Z_TYPE_P(lifetime_or_options) == IS_ARRAY) {
+		int found = 0;
+		zend_string *key;
+		zval *value;
+
+		ZEND_HASH_FOREACH_STR_KEY_VAL(Z_ARRVAL_P(lifetime_or_options), key, value) {
+			if (key) {
+				ZVAL_DEREF(value);
+				if(!strcasecmp("lifetime", ZSTR_VAL(key))) {
+					lifetime = value;
+					convert_to_string_ex(lifetime);
+					found++;
+				} else if(!strcasecmp("path", ZSTR_VAL(key))) {
+					path = Z_STR_P(value);
```

Comment:
You're assuming here that the `path` is a string. Either you need to explicitly check this, or use something like `zval_get_string`.

Reply:
I've tried to use `zval_get_string` for all of these, however, I'm missing something as it is currently leaking on `ext/session/tests/session_set_cookie_params_error.phpt`.

Reply:
AFAIK, you have to `zend_string_release` all strings obtained by using `zval_get_string`.